packages/jazz-react-core/src/hooks.ts

```diff
@@ -138,139 +142,130 @@ export function useCoState<V extends CoValue, D>(
   return value;
 }
 
-export function createUseAccountHooks<Acc extends Account>() {
-  function useAccount(): {
-    me: Acc;
-    logOut: () => void;
-  };
-  function useAccount<D extends DepthsIn<Acc>>(
-    depth: D,
-  ): { me: DeeplyLoaded<Acc, D> | undefined | null; logOut: () => void };
-  function useAccount<D extends DepthsIn<Acc>>(
-    depth?: D,
-  ): { me: Acc | DeeplyLoaded<Acc, D> | undefined | null; logOut: () => void } {
-    const context = useJazzContext<Acc>();
-    const contextManager = useJazzContextManager<Acc>();
-
-    if (!("me" in context)) {
-      throw new Error(
-        "useAccount can't be used in a JazzProvider with auth === 'guest' - consider using useAccountOrGuest()",
-      );
-    }
-
-    const observable = useCoValueObservable<Acc, D>();
-
-    const me = React.useSyncExternalStore<
-      DeeplyLoaded<Acc, D> | undefined | null
-    >(
-      React.useCallback(
-        (callback) => {
-          return subscribeToContextManager(contextManager, () => {
-            const agent = getCurrentAccountFromContextManager(contextManager);
-
-            if (agent._type === "Anonymous") {
-              throw new Error(
-                "useAccount can't be used in a JazzProvider with auth === 'guest' - consider using useAccountOrGuest()",
-              );
-            }
-
-            observable.reset();
-
-            const Schema = agent.constructor as CoValueClass<Acc>;
-
-            return observable
-              .getCurrentObservable()
-              .subscribe(
-                Schema,
-                agent.id,
-                agent,
-                depth ?? ([] as D),
-                callback,
-                callback,
-              );
-          });
-        },
-        [contextManager],
-      ),
-      () => observable.getCurrentValue(),
-      () => observable.getCurrentValue(),
-    );
+function useAccount<A extends RegisteredAccount>(): {
+  me: A;
+  logOut: () => void;
+};
+function useAccount<A extends RegisteredAccount, D extends DepthsIn<A>>(
+  depth: D,
+): { me: DeeplyLoaded<A, D> | undefined | null; logOut: () => void };
+function useAccount<A extends RegisteredAccount, D extends DepthsIn<A>>(
+  depth?: D,
+): { me: A | DeeplyLoaded<A, D> | undefined | null; logOut: () => void } {
+  const context = useJazzContext<A>();
+  const contextManager = useJazzContextManager<A>();
```

Comment:
Moved away from the factory pattern because the RegisteredAccount injection was broken by the `d.ts` compilation.

Now the jazz-react and jazz-react-native simply re-export this hook while registering the RegisteredAccount type.

This way the user can still do the declaration on these two packages and reach the RegisteredAccount type here